base/gnfd/gnfd_service.go

```diff
@@ -783,3 +787,70 @@ func (g *Gnfd) VerifyPutObjectPermission(ctx context.Context, account, bucket, o
 	}
 	return false, err
 }
+
+// ConfirmTransaction is used to confirm whether the transaction is on the chain.
+func (g *Gnfd) ConfirmTransaction(ctx context.Context, txHash string) (*sdk.TxResponse, error) {
```

Comment:
It should have a timeout for retry.

Reply:
Currently, only retry once and there is a timeout in waitForNextBlock.

Reply:
Why only retry once?

Reply:
Retrying once basically guarantees whether it is in the current block or the next block.
what is the suggestion here.

Reply:
At least 3 blocks.

Reply:
Fixed.